tests/components/tesla_wall_connector/test_init.py

```diff
@@ -0,0 +1,39 @@
+"""Test the Tesla Wall Connector config flow."""
+from tesla_wall_connector.exceptions import WallConnectorConnectionError
+
+from homeassistant import config_entries
+from homeassistant.components.tesla_wall_connector.const import DOMAIN
+from homeassistant.core import HomeAssistant
+
+from .conftest import create_wall_connector_entry
+
+
+async def test_init_success(hass: HomeAssistant) -> None:
+    """Test setup and that we get the device info, including firmware version."""
+
+    entry = await create_wall_connector_entry(hass)
+
+    assert entry.state == config_entries.ConfigEntryState.LOADED
+    assert hass.data[DOMAIN]
+    assert hass.data[DOMAIN][entry.entry_id]
```

Comment:
```suggestion
```


Please don't access internals (`hass.data`) in tests https://developers.home-assistant.io/docs/development_testing/#writing-tests-for-integrations

Reply:
Ok... this however seems to be a very common practice in tests for other integrations. Should I just skip checking that a failed setup of the integration does not pollute the hass.data space?

Reply:
We are trying to move away from that.  The code base is 8 years old so there is still quite a bit of legacy. 

Good to drop the check.

Reply:
Ok. Fixed in 776b881